### PR TITLE
docs: Refine READMEs

### PR DESCRIPTION
**Pull request by jhaaaa**
Updates to READMEs based on some UXR feedback.

### CODE REVIEW

File: packages/react-sdk/README.md [hunk 1]
```diff
@@ -691,7 +685,7 @@ Run `yarn dev` to build the SDK on changes and launch Storybook.
 
 - `yarn build`: Builds the SDK
 - `yarn clean`: Removes `node_modules`, `lib`, and `.turbo` folders
-- `yarn dev`: Launches Storybook for SDK components
+- `yarn dev`: Launches Storybook for SDK components. Also launches the React Quickstart example app at `http://127.0.0.1:5173/`.
```
Comment (jhaaaa) [inline packages/react-sdk/README.md:688]: @rygine - I have a few questions about the `yarn dev` command when run in different directories. =) I wonder if I'm misunderstanding the command? Thank you....

- When I run `yarn dev` in `xmtp-web` - what should I expect? What I see is that it starts the example app at http://127.0.0.1:5173/, but does it also start Storybook? It doesn't automatically open a Storybook browser tab for me - should it? What port is it running on?

- When I run `yarn dev` in `xmtp-web/packages/react-sdk` - what should I expect? It doesn't open Storybook and I don't see the example app running.

- When I run `yarn dev` in `xmtp-web/examples/react-quickstart` - it starts the example app at http://127.0.0.1:5173/.

File: packages/react-sdk/README.md [hunk 2]
```diff
@@ -691,7 +685,8 @@ Run `yarn dev` to build the SDK on changes and launch Storybook.
 
 - `yarn build`: Builds the SDK
 - `yarn clean`: Removes `node_modules`, `lib`, and `.turbo` folders
-- `yarn dev`: Launches Storybook for SDK components
+- `yarn dev`: Develop the `react-sdk` package
+- `yarn storybook`: Launches Storybook for SDK components at `http://localhost:6006`
```
Comment (jhaaaa) [inline packages/react-sdk/README.md:689]: Thank you for this @rygine - ran it and it works for me. 

I wrote the description for `yarn dev` based on the `yarn dev` description for the `xmtp-web` level README. Is this correct? What does it mean?

File: examples/react-quickstart/README.md [hunk 3]
```diff
@@ -14,17 +14,17 @@ To learn more about XMTP and get answers to frequently asked questions, see [FAQ
...
 
-After installing the dependencies, run `yarn dev` to start developing.
+Install the [React client SDK and requirements](/packages/react-sdk/README.md) and then run `yarn dev` to start developing.
 
 ## Useful commands
 
 - `yarn build`: Builds the example app
 - `yarn clean`: Removes `node_modules`, `dist`, and `.turbo` folders
-- `yarn dev`: Launches the example app in development mode
+- `yarn dev`: Launches the example app in development mode at `http://127.0.0.1:5173`
```
Comment (rygine) [inline examples/react-quickstart/README.md:27]: i don't think it's necessary to add the IP address and port here. it's in the terminal when running `yarn dev` and the port can be changed.

File: examples/react-quickstart/README.md [hunk 4]
```diff
@@ -14,17 +14,17 @@ To learn more about XMTP and get answers to frequently asked questions, see [FAQ
 
 ### Limitations
 
-This example app demonstrates features of the React SDK, but it isn't a complete solution. For example, the list of conversations doesn't update when new messages arrive in existing conversations.
+This example app demonstrates features of the React XMTP client SDK, but it isn't a complete solution. For example, the list of conversations doesn't update when new messages arrive in existing conversations.
 
 ## Developing
 
-After installing the dependencies, run `yarn dev` to start developing.
+Install the [React client SDK and requirements](/packages/react-sdk/README.md) and then run `yarn dev` to start developing.
```
Comment (rygine) [inline examples/react-quickstart/README.md:21]: it's not so much "install" as it is "build" that needs to occur prior to running `yarn dev` inside of this folder. running `yarn dev` at the repo root will handle this automatically.

to build manually, you need to go to the `react-sdk` folder and run `yarn build` first.

File: packages/react-sdk/README.md [hunk 5]
```diff
@@ -28,17 +28,11 @@ These ready-made components can help you quickly build a chat app.
 
 - Node 16.10+
 - React 16.14+
+- Yarn v3+. See [Yarn Installation](https://yarnpkg.com/getting-started/install).
```
Comment (rygine) [inline packages/react-sdk/README.md:31]: this is not a requirement for use of the SDK outside of this repository. sorry for the confusion here.

File: packages/react-sdk/README.md (hunk 2 above)
Comment (rygine) [reply packages/react-sdk/README.md:689]: `yarn dev` will build the SDK and watch for code changes, which will trigger a re-build.

File: packages/react-sdk/README.md (hunk 5 above)
Comment (jhaaaa) [reply packages/react-sdk/README.md:31]: Oh my gosh, no worries at all Ry - I am sure I made some assumptions and miscommunicated in our convo - thank you for the clarification!

File: packages/react-sdk/README.md [hunk 6]
```diff
@@ -28,6 +28,7 @@ These ready-made components can help you quickly build a chat app.
 
 - Node 16.10+
 - React 16.14+
+- For use within this repo, Yarn v3+. See [Yarn Installation](https://yarnpkg.com/getting-started/install).
```
Comment (rygine) [inline packages/react-sdk/README.md:31]: sorry again for this confusion on this. Yarn V3 is only required at the repo root level and when developing examples and packages in this repo. all external consumption of the React SDK does not have this requirement.

File: packages/react-sdk/README.md [hunk 7]
```diff
@@ -38,7 +39,7 @@ npm install @xmtp/react-sdk@preview
 # pnpm
 pnpm install @xmtp/react-sdk@preview
 
-# yarn
+# for use within this repo, yarn
```
Comment (rygine) [inline packages/react-sdk/README.md:42]: see comment above

File: packages/react-sdk/README.md (hunk 6 above)
Comment (jhaaaa) [inline packages/react-sdk/README.md:31]: Ry, do these clarifications above and below - about yarn being a requirement for use within this repo only - make sense? Did I interpret this correctly? TY!

File: README.md [hunk 8]
```diff
@@ -30,7 +30,7 @@ After installing the requirements, run `yarn && yarn dev` to start developing.
 
 - `yarn build`: Build all packages
 - `yarn clean`: Remove all `node_modules`, `.turbo`, and build folders, clear Yarn cache
-- `yarn dev`: Develop all packages and examples
+- `yarn dev`: Builds all packages and examples and watches for changes, which will trigger rebuilds
```
Comment (rygine) [inline README.md:33]: this just builds packages/react-sdk, then runs the `dev` command in all other packages and examples. currently, it builds the SDK, runs the example app in dev mode, and starts dev mode on the SDK.

File: README.md [hunk 9]
```diff
@@ -18,9 +18,9 @@ To learn more about the contents of this repository, see this README and the REA
 
 ## Requirements
 
-This repository requires Yarn v3+ and Node 16.10+.
-
-To install Yarn v3, see [Yarn Installation](https://yarnpkg.com/getting-started/install).
+- Node 16.10+
+- React 16.14+
+- Yarn v3+ is required only at the root level of this repo and when developing examples and packages in this repo. See [Yarn Installation](https://yarnpkg.com/getting-started/install). External consumption of the React SDK does not require Yarn.
```
Comment (rygine) [inline README.md:23]: WDYT of something like this:

```
Yarn v3+ is only required when working with this repo. See [Yarn Installation](https://yarnpkg.com/getting-started/install).
```

the reason i think this is OK is the nature of package managers and how developers interact with them. it's known that you can use any package manager to consume packages on NPM, but may need to use a specific one when working within a repo.

File: packages/react-sdk/README.md [hunk 10]
```diff
@@ -28,6 +28,7 @@ These ready-made components can help you quickly build a chat app.
 
 - Node 16.10+
 - React 16.14+
+- Yarn v3+ is required only at the root level of this repo and when developing examples and packages in this repo. See [Yarn Installation](https://yarnpkg.com/getting-started/install). External consumption of the React SDK does not require Yarn.
```
Comment (rygine) [inline packages/react-sdk/README.md:31]: i think we can remove this line

**Review by rygine — APPROVED** [review]

**Pull request merged**